keel-orca/src/main/kotlin/com/netflix/spinnaker/keel/orca/OrcaTaskMonitorAgent.kt

```diff
@@ -75,13 +75,23 @@ class OrcaTaskMonitorAgent(
     }
   }
 
-    // make sure it's only 1 context per run
-    private fun List<Map<String, Any>>?.getFailureMessage(): String? {
-      if (this.isNullOrEmpty()) {
-        return ""
+  // get the exception - can be either general orca exception or kato specific
+  private fun List<Map<String, Any>>?.getFailureMessage(): String? {
+
+    this?.forEach { it ->
+      val context: OrcaContext? = it["context"]?.let { mapper.convertValue(it) }
+
+      // find the first exception and return
+      if (context?.exception != null) {
+        return context.exception.details.errors.joinToString(",")
       }
-      val context: OrcaContext? = this.first()["context"]?.let { mapper.convertValue(it) }
 
-      return context?.exception?.details?.errors?.joinToString(",")
+      if (context?.katoException != null) {
+        val katoError: KatoException? = context.katoException.first()["exception"]?.let { mapper.convertValue(it) }
+        return katoError?.message
+      }
     }
+
+    return ""
```

Comment:
hmm not sure what you mean - should I some more checks here to make sure `null` will not be returned?

Reply:
The method signature has a return type of `String?` but here it looks like the fallback is to return `""`. I'm not clear if there are situations where it can return `null` and what the difference is between `null` and `""`.

Reply:
Agreed, will change it into `null`.